services/gateway/proxy/src/main/java/org/eclipse/ditto/services/gateway/proxy/actors/AbstractPolicyEnforcerActor.java

```diff
@@ -190,10 +188,26 @@ protected boolean isEnforcerAvailable() {
 
     protected <T extends WithDittoHeaders> T enrichDittoHeaders(final WithDittoHeaders<T> withDittoHeaders,
```

Comment:
IMHO it would be better to replace `WithDittoHeaders` with `Signal`. Then the ugly `instanceof` checks could be replaced with a check for the Signal type like for example `if (CreatePolicy.TYPE == signal.getType() ...)`.

Reply:
ah, good point - this is a relict from where `CommanResponses` were not yet a `Signal`

Reply:
but regarding the `instanceof` checks vs a `CreatePolicy.TYPE.equals(signal.getType())` i personally think the instanceof check is clearer

Reply:
Well, this is personal perception and thus, for you, this might be true. `instanceof` is a technical feature of Java. In this case we are dealing with domain knowledge ("What signal type I am handling right now?") and thus I prefer using the means of the domain (aka `TYPE` and `getType()`) for solving the problem.

Reply:
understood - but in this case I also cast to the type in order to get the "policy" out of the commands - therefore the technical check for instance is also relevant :)